numpy/lib/function_base.py

```diff
@@ -1526,16 +1530,30 @@ def unwrap(p, discont=pi, axis=-1):
     array([ 0.        ,  0.78539816,  1.57079633,  5.49778714,  6.28318531]) # may vary
     >>> np.unwrap(phase)
     array([ 0.        ,  0.78539816,  1.57079633, -0.78539816,  0.        ]) # may vary
-
+    >>> unwrap([0, 1, 2, -1, 0], period=4)
+    array([0., 1., 2., 3., 4.])
+    >>> unwrap([ 1, 2, 3, 4, 5, 6, 1, 2, 3], period=6)
+    array([1., 2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> unwrap([2, 3, 4, 5, 2, 3, 4, 5], period=4)
+    array([2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> phase_deg = np.mod(np.linspace(0,720,19), 360) - 180
+    >>> unwrap(phase_deg, period=360)
+    array([-180., -140., -100.,  -60.,  -20.,   20.,   60.,  100.,  140.,
+        180.,  220.,  260.,  300.,  340.,  380.,  420.,  460.,  500.,
+        540.])
     """
     p = asarray(p)
     nd = p.ndim
     dd = diff(p, axis=axis)
+    if discont is None:
+        discont = period/2
     slice1 = [slice(None, None)]*nd     # full slices
     slice1[axis] = slice(1, None)
     slice1 = tuple(slice1)
-    ddmod = mod(dd + pi, 2*pi) - pi
-    _nx.copyto(ddmod, pi, where=(ddmod == -pi) & (dd > 0))
+    ddmod = mod(dd + period/2, period) - period/2
+     # for `mask = (abs(dd) == period/2)`, the above line made `ddmod[mask] == -period/2`.
+    # correct these such that `ddmod[mask] == sign(dd[mask])*period/2`.
+    _nx.copyto(ddmod, period/2, where=(ddmod == -period/2) & (dd > 0))
```

Comment:
```suggestion
Perhaps to support integers:
```suggestion
    dtype = np.result_type(dd, period)
    if _nx.isubdtype(dtype, _nx.integer):
        interval_low = (-period) // 2
        interval_high = interval_low + period
    else:
        interval_low = -period / 2
        interval_high = +interval_low
    ddmod = mod(dd - interval_low, period) + interval_low
    # for `mask = (abs(dd) == period/2)`, the above line made `ddmod[mask] == -period/2`.
    # correct these such that `ddmod[mask] == sign(dd[mask])*period/2`.
    _nx.copyto(ddmod, interval_high, where=(ddmod == interval_low) & (dd > 0))
```

Reply:
aaaah, thanks! I was looking for a function like `_nx.issubdtype` for quite a while. `type hierarchy` is the word I was missing. Thanks!
I also like the names `interval_low` and `_high`. They describe better what is going on there.

Reply:
It's not working properly yet. We have again the problem that whether or not a jump is corrected, depends on the sign. In the first case the jump is `-4`, in the second it's `4`.

>>> a= np.concatenate((np.arange(0,5), np.arange(0,7)))
>>> a
array([0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 5, 6])
>>> unwrap(a, period=7)
# dd:    [ 1  1  1  1 -4  1  1  1  1  1  1]
# ddmod:  [ 1  1  1  1 -4  1  1  1  1  1  1]
array([0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 5, 6])
>>> unwrap(a[::-1], period=7)
# dd:    [-1 -1 -1 -1 -1 -1  4 -1 -1 -1 -1]
# ddmod:  [-1 -1 -1 -1 -1 -1 -3 -1 -1 -1 -1]
array([ 6,  5,  4,  3,  2,  1,  0, -3, -4, -5, -6, -7])

Reply:
The output of `unwrap(a, period=7)` is wrong there, right?

Reply:
What happens if you add `interval_high = -interval_low` for the integer case?

Reply:
> The output of `unwrap(a, period=7)` is wrong there, right?

Yes, in both cases the jump of `4` is larger than `period/2`.

Reply:
Ok, I thank I have it now. Checking for `period % 2 == 0` is the only way, I think. Tested it with a `period` of 6 and 7, and jumps of 3 and 4. They worked as desired.